### ISSUE TITLE
Flaky-test: Failed to get tenant admin data for tenant public

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#15652]
There's a common problem in tests which make them flaky "Failed to get tenant admin data for tenant public" happens when creating a namespace.

[example failure](https://github.com/apache/pulsar/runs/6484025367?check_suite_focus=true#step:12:17310)

```
2022-05-18T06:44:24,018 - INFO  - [docker-java-stream--110056506:DockerUtils$2@252] - DOCKER.exec(ReaderMessagingTest-cpgsi-pulsar-broker-1:/pulsar/bin/pulsar-admin namespaces create public/ns-ivorhkkv --clusters ReaderMessagingTest-cpgsi): STDOUT: 2022-05-18T06:44:24,012+0000 [AsyncHttpClient-7-1] WARN  org.apache.pulsar.client.admin.internal.BaseResource - [http://localhost:8080/admin/v2/namespaces/public/ns-ivorhkkv] Failed to perform http put request: javax.ws.rs.InternalServerErrorException: HTTP 500 Failed to get data from /admin/policies/public
  2022-05-18T06:44:24,079 - INFO  - [docker-java-stream--110056506:DockerUtils$2@252] - DOCKER.exec(ReaderMessagingTest-cpgsi-pulsar-broker-1:/pulsar/bin/pulsar-admin namespaces create public/ns-ivorhkkv --clusters ReaderMessagingTest-cpgsi): STDERR: --- An unexpected error occurred in the server ---
  
  Message: Failed to get data from /admin/policies/public
  
  Stacktrace:
  
  org.apache.pulsar.metadata.api.MetadataStoreException: Failed to get data from /admin/policies/public
  	at org.apache.pulsar.broker.resources.BaseResources.get(BaseResources.java:88)
  	at org.apache.pulsar.broker.resources.TenantResources.getTenant(TenantResources.java:62)
  	at org.apache.pulsar.broker.web.PulsarWebResource.validateClusterForTenant(PulsarWebResource.java:401)
  	at org.apache.pulsar.broker.admin.impl.NamespacesBase.lambda$validatePolicies$89(NamespacesBase.java:2068)
  	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
  	at org.apache.pulsar.broker.admin.impl.NamespacesBase.validatePolicies(NamespacesBase.java:2068)
  	at org.apache.pulsar.broker.admin.impl.NamespacesBase.lambda$internalCreateNamespace$3(NamespacesBase.java:136)
  	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718)
  	at java.base/java.util.concurrent.CompletableF
  2022-05-18T06:44:24,079 - INFO  - [docker-java-stream--110056506:DockerUtils$2@252] - DOCKER.exec(ReaderMessagingTest-cpgsi-pulsar-broker-1:/pulsar/bin/pulsar-admin namespaces create public/ns-ivorhkkv --clusters ReaderMessagingTest-cpgsi): STDERR: uture.postComplete(CompletableFuture.java:510)
  	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
  	at org.apache.pulsar.metadata.impl.ZKMetadataStore.lambda$existsFromStore$8(ZKMetadataStore.java:315)
  	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
  	at java.base/java.lang.Thread.run(Thread.java:833)
  Caused by: java.util.concurrent.TimeoutException
  	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
  	at java.base/java.util.concurr
```

**Comment by lhotari** [comment]
Similar problem in a completely different test. 
```
2022-05-18T07:03:48,574+0000 [metadata-store-12-1] ERROR org.apache.pulsar.broker.web.PulsarWebResource - Failed to get tenant admin data for tenant public
org.apache.pulsar.metadata.api.MetadataStoreException: Failed to get data from /admin/policies/public
	at org.apache.pulsar.broker.resources.BaseResources.get(BaseResources.java:88) ~[org.apache.pulsar-pulsar-broker-common-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at org.apache.pulsar.broker.resources.TenantResources.getTenant(TenantResources.java:62) ~[org.apache.pulsar-pulsar-broker-common-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at org.apache.pulsar.broker.web.PulsarWebResource.validateClusterForTenant(PulsarWebResource.java:401) ~[org.apache.pulsar-pulsar-broker-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at org.apache.pulsar.broker.admin.impl.NamespacesBase.lambda$validatePolicies$89(NamespacesBase.java:2068) ~[org.apache.pulsar-pulsar-broker-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at org.apache.pulsar.broker.admin.impl.NamespacesBase.validatePolicies(NamespacesBase.java:2068) ~[org.apache.pulsar-pulsar-broker-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at org.apache.pulsar.broker.admin.impl.NamespacesBase.lambda$internalCreateNamespace$3(NamespacesBase.java:136) ~[org.apache.pulsar-pulsar-broker-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.lambda$existsFromStore$8(ZKMetadataStore.java:315) ~[org.apache.pulsar-pulsar-metadata-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[io.netty-netty-common-4.1.77.Final.jar:4.1.77.Final]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.util.concurrent.TimeoutException
	at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095) ~[?:?]
	at org.apache.pulsar.broker.resources.BaseResources.get(BaseResources.java:83) ~[org.apache.pulsar-pulsar-broker-common-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	... 17 more
2022-05-18T07:03:48,578+0000 [metadata-store-12-1] ERROR org.apache.pulsar.broker.admin.v2.Namespaces - [null] Failed to create namespace public/ns-vxxptgiy
java.util.concurrent.CompletionException: org.apache.pulsar.broker.web.RestException: HTTP 500 Failed to get data from /admin/policies/public
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[?:?]
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.lambda$existsFromStore$8(ZKMetadataStore.java:315) ~[org.apache.pulsar-pulsar-metadata-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[io.netty-netty-common-4.1.77.Final.jar:4.1.77.Final]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: org.apache.pulsar.broker.web.RestException: HTTP 500 Failed to get data from /admin/policies/public
	at org.apache.pulsar.broker.web.PulsarWebResource.validateClusterForTenant(PulsarWebResource.java:408) ~[org.apache.pulsar-pulsar-broker-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at org.apache.pulsar.broker.admin.impl.NamespacesBase.lambda$validatePolicies$89(NamespacesBase.java:2068) ~[org.apache.pulsar-pulsar-broker-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at org.apache.pulsar.broker.admin.impl.NamespacesBase.validatePolicies(NamespacesBase.java:2068) ~[org.apache.pulsar-pulsar-broker-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at org.apache.pulsar.broker.admin.impl.NamespacesBase.lambda$internalCreateNamespace$3(NamespacesBase.java:136) ~[org.apache.pulsar-pulsar-broker-2.11.0-SNAPSHOT.jar:2.11.0-SNAPSHOT]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718) ~[?:?]
	... 10 more
```

**Comment by lhotari** [comment]
@Technoboy- Have you seen these errors? asking, since #15518 changes seem in the same area where these timeouts happen.

**Comment by lhotari** [comment]
3 integration/system test failures in a single workflow run, all caused by the same issue: 
https://github.com/apache/pulsar/runs/6484025367?check_suite_focus=true
https://github.com/apache/pulsar/runs/6484269027?check_suite_focus=true
https://github.com/apache/pulsar/runs/6484269298?check_suite_focus=true

**Comment by lhotari** [comment]
Another one, https://github.com/apache/pulsar/runs/6485278752?check_suite_focus=true

**Comment by lhotari** [comment]
#15603 might be one of the changes to check as well.

I'm running an experiment where I revert both #15603 and #15518 to see if that makes a difference.

**Comment by lhotari** [comment]
Yet another flaky failure from another PR, https://github.com/apache/pulsar/runs/6463528996?check_suite_focus=true . That failed in @wolfstudy 's PR #15628 .

**Comment by Technoboy-** [comment]
Make a fix here https://github.com/apache/pulsar/pull/15605/commits/d2c0fc8360cbdf247921d540a1ee2dc64b6bcfe9

**Comment by lhotari** [comment]
> Make a fix here [d2c0fc8](https://github.com/apache/pulsar/commit/d2c0fc8360cbdf247921d540a1ee2dc64b6bcfe9)

@Technoboy- looks good, is that already in a PR?

**Comment by Technoboy-** [comment]
> > Make a fix here [d2c0fc8](https://github.com/apache/pulsar/commit/d2c0fc8360cbdf247921d540a1ee2dc64b6bcfe9)
> 
> @Technoboy- looks good, is that already in a PR?

It contains in https://github.com/apache/pulsar/pull/15605.
